src/browser_launchers/puppeteer_launcher.js

```diff
@@ -124,6 +48,73 @@ export function apifyOptionsToLaunchOptions(launchContext) {
  *   Using this configuration, you can disable some of the hiding tricks.
  *   For these settings to take effect `stealth` must be set to true
  */
+export class PuppeteerLauncher extends BrowserLauncher {
+    static optionsShape = {
+        ...BrowserLauncher.optionsShape,
+        launcher: ow.optional.object,
+        userAgent: ow.optional.string,
+        stealth: ow.optional.boolean,
+        stealthOptions: ow.optional.object,
+    }
+
+    /**
+    * @param {PuppeteerLaunchContext} launchContext
+    * All `PuppeteerLauncher` parameters are passed via an launchContext object.
+    */
+    constructor(launchContext = {}) {
+        ow(launchContext, 'PuppeteerLauncher', ow.object.exactShape(PuppeteerLauncher.optionsShape));
+
+        launchContext.launcher = launchContext.launcher || require('puppeteer'); // eslint-disable-line
+        const {
+            userAgent,
+            stealth,
+            stealthOptions,
+            ...browserLauncherOptions
+        } = launchContext;
+
+        super(browserLauncherOptions);
+
+        this.userAgent = userAgent;
+        this.stealth = stealth; // @TODO: time to rename to useStealth?
+        this.stealthOptions = stealthOptions;
+
+        this.Plugin = PuppeteerPlugin;
+    }
+
+    async launch() {
+        const browser = await super.launch();
+
+        if (this.stealth) {
+            applyStealthToBrowser(browser, this.stealthOptions);
+        }
```

Comment:
We better not forget about this. Can you create an issue please?

Reply:
Added #901.